packages/gatsby/src/redux/actions/public.js

```diff
@@ -447,6 +448,33 @@ ${reservedFields.map(f => `  * "${f}"`).join(`\n`)}
     )
   }
 
+  if (state.queries.createPagesFinished) {
```

Comment:
We need to know wether this is from `createPages` or `createPagesStatefully` (reason the previous code specifically targeted `createPages` API). `createPagesStatefully` is the escape hatch from the "fully managed" mode - right now this is only used by `gatsby-plugin-page-creator` (IIRC), and allows the create/delete pages in response to FS events which wouldn't trigger `createPages` rerun.

Seems like we already can know wether it was called "from" `*Statefully` hook - https://github.com/gatsbyjs/gatsby/blob/064207dffdee9d604f92825ae9569309e70fcb9f/packages/gatsby/src/redux/actions/public.js#L401-L402

So we should probably use that and only show the warning if the `action` is not coming from `createPagesStatefully`